src/t_string.c

```diff
@@ -73,7 +73,9 @@ void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire,
         if (getGenericCommand(c) == C_ERR) return;
     }
 
-    found = (lookupKeyWrite(c->db,key) != NULL);
+    dictEntry *de = dbFind(c->db, key->ptr);
+    robj *o = lookupKeyWriteWithDictEntry(c->db,key,de);
```

Comment:
it's nice that we save a dictFind in case the key was found. but we don't save it when the key wasn't fount (we pass NULL and try again).
it would be good idea to design an interface where we can pass an optional argument that can also represent a not-found case.
maybe a reference to a dictEntry?

Reply:
btw, maybe another solution for all of that is a lookupKey variant that has an additional return value and returns the dictEntry?